scripts/kb.py

```diff
@@ -0,0 +1,633 @@
+        Retrieve candidate entities for specified mention. If no candidate is found for a given mention, an empty list
+        is returned.
+        mention (Span): Mention for which to get candidates.
+        RETURNS (Iterable[Candidate]): Identified candidates.
+        """
+        return next(iter(next(self.get_candidates_all([mention]))))
+
+    def _get_vectors(self, rowids: Iterable[int]) -> Iterable[Iterable[float]]:
+        """
+        Return vectors for entities.
+        rowids (Iterable[int]): ROWID values for entities in table `entities`.
+        RETURNS (Iterable[Iterable[float]]): Vectors for specified entities.
+        """
+        # Annoy doesn't seem to offer batched retrieval.
+        return [self._annoy.get_item_vector(rowid - 1) for rowid in rowids]
+
+    def get_vectors(self, qids: Iterable[str]) -> Iterable[Iterable[float]]:
+        """
+        Return vectors for entities.
+        qids (str): Wiki QIDs.
+        RETURNS (Iterable[Iterable[float]]): Vectors for specified entities.
+        """
+        if not isinstance(qids, Sized):
+            qids = set(qids)
+
+        # Fetch row IDs for QIDs, resolve to vectors in Annoy index.
+        return self._get_vectors(
+            [
+                row["ROWID"]
+                for row in self._db_conn.cursor()
+                .execute(
+                    "SELECT ROWID FROM entities WHERE id in (%s)"
+                    % ",".join("?" * len(qids)),
+                    tuple(qids),
+                )
+                .fetchall()
+            ]
+        )
+
+    def get_vector(self, qid: str) -> Iterable[float]:
+        """
+        Return vector for qid.
+        qid (str): Wiki QID.
+        RETURNS (Iterable[float]): Vector for specified entities.
+        """
+        return next(iter(self.get_vectors([qid])))
+
+    @staticmethod
+    def _hash_file(file_path: Path, blocksize: int = 2**20) -> str:
+        """Generates MD5 file of hash iteratively (without loading entire file into memory).
+        Source: https://stackoverflow.com/a/1131255.
+        file_path (Path): Path of file to hash.
+        blocksize (int): Size of blocks to load into memory (in bytes).
+        RETURN (str): MD5 hash of file.
+        """
+        file_hash = hashlib.md5()
+        with open(file_path, "rb") as f:
+            buf = f.read(blocksize)
+            while buf:
+                file_hash.update(buf)
+                buf = f.read(blocksize)
+        return file_hash.hexdigest()
+
+    def _fetch_candidates_by_alias(
+        self, mentions: Tuple[Span, ...]
+    ) -> Dict[str, List[Dict[str, Union[str, int, float]]]]:
+        """Fetches candidates for mentions by fuzzily matching aliases to the mentions.
+        mentions (Tuple[Span, ..]): List of mentions for which to fetch candidates.
+        RETURN List[Dict[str, Dict[str, Union[str, int, float]]]]: List of candidates per mention, sorted by distance
+            to mention, (3) occurence count in Wikipedia.
```

Comment:
is there a 2) missing from the docstring here?

Reply:
Updated in https://github.com/explosion/wikid/pull/4/commits/33d6db297cb5d8bf9ee85023c604b8f9e236cd3e.